### PR TITLE
fn: improve funcfile doc

### LINKED ISSUE #366: Documentation lagging behind after Hot Containers
https://github.com/iron-io/functions/blob/master/docs/function-file.md is outdated versus https://github.com/iron-io/functions/blob/master/fn/funcfile.go

### PR DESCRIPTION
**Pull request by ccirello**
Fixes #366

### CODE REVIEW

**Comment by ccirello** [comment]
@pedronasser @seiflotfy @treeder

File: docs/function-file.md [hunk 1]
```diff
@@ -41,10 +41,26 @@ Default: `sync`.
...
 
+`headers` (optional) is a set of HTTP headers to be returned in the response of
+this function calls.
+
 `config` (optional) is a set of configurations to be passed onto the route
 setup. These configuration options shall override application configuration
 during functions execution.
 
 `build` (optional) is an array of shell calls which are used to helping building
 the image. These calls are executed before `fn` calls `docker build` and
```
Comment (treeder) [inline docs/function-file.md]: Should these say `fn build` and `fn push` instead of docker?

Comment (ccirello) [inline docs/function-file.md]: Build is used in of `fn build`. If we put `fn build` then it would become a recursive explanation.

Comment (ccirello) [inline docs/function-file.md]: done

**Comment by seiflotfy** [comment]
looks good, please resolve conflict

**Pull request merged**